### DISCUSSION TITLE
Question from a new user on pasting builds

### DISCUSSION THREAD

**Discussion by Little-Vixly** [maruohon/litematica#520] [Q&A]
Ok so this is my first time using litematica and I'm *slowly* learning. I'm working on a creative server and I want to paste a big tree into the world, this is allowed on the server since it's creative. It's too big of a tree to where it lags me out anytime I try to paste the schematic is there any way to paste it in separate sections (like mini sections that make a whole) so I stop getting kicked? Also if the answer is yes can you please explain how like I said I'm learning *slowly*

**Comment by maruohon** [comment]
If you haven't touched any paste related settings yet, then I'd recommend first of all lowering the `commandLimitPerTick` config in the Generic category. The default is 64 commands per tick which is usually too much. Try 16 or 8 or 4 or even 2, it also depends on the server settings, and for example some Spigot/Paper based server have really strict packet rate limits. On vanilla or Fabric servers 24 and 32 are usually still fine, even 64 might be fine. There is also the `commandTaskInterval` config, which controls if the task runs every tick or at a larger interval.

As for pasting the build in smaller parts, the Paste in multiplayer does respect the render layer range currently, so you can use the render layers to limit which part of it gets pasted (`M + C` -> Render Layers). Depending on the shape switching the axis in the render layers setting from the normal Y to X or Z might also be useful in this, and probably using the Layer Range setting.

If the render layers setting is not enough, then you could also use sub-regions to only paste smaller parts of the entire schematic at once, but that of course would require the schematic to have been saved using multiple regions in the first place. You can also split it up by saving smaller parts of it as separate schematics, but that just gets into rather annoying territory with lots of fiddling around saving multiple schematics and positioning them correctly etc. Usually using the lower command rate limit is enough, and maybe additionally using the render layers to limit the range of a single paste operation.

**Reply by Little-Vixly** [reply]
So for the second paragraph here the render layers, when I go to paste it still pastes the whole thing regardless of which layer it is showing. Am I missing something here? Thank you!

**Reply by maruohon** [reply]
Oh are you not on a server? That feature currently only works when pasting on a server. And also I have to say I don't even remember if it works with the new fill-based pasting... There is an option in the Generic configs to disable the fill-based pasting mode that was added recently. But on the other hand if your problem was mostly the command spam, then the fill-based paste mode is something that you really would want to keep enabled, as it cuts down on the command spam significantly, by using `/fill` commands where possible, instead of individual `/setblock` commands for *everything* even in large volumes of the same block.
